### PR TITLE
Adding end time and run duration to Compare Runs view

### PR DESCRIPTION
**Pull request by RealArpanBhattacharya**
Signed-off-by: Arpan Bhattacharya <tell.arpan.bhattacharya@gmail.com>

## What changes are proposed in this pull request?

In the "Compare Runs" section - display end time and run duration. 
Addresses [this](https://github.com/mlflow/mlflow/issues/2029) issue.

## How is this patch tested?

Still trying to figure out the automated test suite. 
Screenshot:
image

### Is this a user-facing change?

It is. In the "Compare Runs" section of the tracking UI, the end timestamp, and the duration of the experiment will be shown upon pushing this patch.

### What component(s), interfaces, languages, and integrations does this PR affect?

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section

### CODE REVIEW

**Comment by harupy** [comment]
@RealArpanBhattacharya Thanks for the PR and sorry for the very late reply :)

**Comment by harupy** [comment]
image

**Comment by harupy** [comment]
Fixed

**Review by BenWilson2 — APPROVED** [review]
LGTM! Useful addition

File: mlflow/server/js/src/experiment-tracking/components/CompareRunView.js [hunk 1]
```diff
@@ -267,6 +267,68 @@ export class CompareRunView extends Component {
...
 
+  renderTimeRows(colWidthStyle) {
+    const getTimeAttributes = (runInfo) => {
+      const startTime = runInfo.getStartTime();
+      const endTime = runInfo.getEndTime();
+      return {
+        runUuid: runInfo.run_uuid,
+        startTime: startTime ? Utils.formatTimestamp(startTime) : '(unknown)',
+        endTime: endTime ? Utils.formatTimestamp(endTime) : '(unknown)',
+        duration: startTime && endTime ? Utils.getDuration(startTime, endTime) : '(unknown)',
```
Comment (sunishsheth2009) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:278]: Can we translate the `(unknown)` string as well using formatMessage?

File: mlflow/server/js/src/experiment-tracking/components/CompareRunView.js [hunk 2]
```diff
@@ -267,6 +267,68 @@ export class CompareRunView extends Component {
...
+              defaultMessage='Duration:'
+              // eslint-disable-next-line max-len
+              description='Row title for the duration of runs on the experiment compare runs page'
+            />
+          </th>
+          {runTimeAttributes.map(({ runUuid, duration }) => [runUuid, duration]).map(renderCell)}
+        </tr>
+      </>
+    );
+  }
```
Comment (sunishsheth2009) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:330]: Can we dry the duplicate code out?

We can create an array with all the configs and loop over the array to render each table row. What do you think?


example:
```
const tableCells = [
   {
       translation = <FormattedMessage
              defaultMessage='Start Time:'
              // eslint-disable-next-line max-len
              description='Row title for the start time of runs on the experiment compare runs page'
            />
   },
   {
         translation =   <FormattedMessage
              defaultMessage='End Time:'
              // eslint-disable-next-line max-len
              description='Row title for the end time of runs on the experiment compare runs page'
            />
   }
   ....
]
```


usage:

```
tableCells.map((cell) => {
<tr>
          <th scope='row' className='head-value sticky-header' style={colWidthStyle}>
            {cell.translation}
          </th>
          {runTimeAttributes.map(({ runUuid, startTime }) => [runUuid, startTime]).map(renderCell)}
        </tr>

})
```

Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/CompareRunView.js:330]: sounds good!

**Review by sunishsheth2009 — APPROVED** [review]

File: mlflow/server/js/src/i18n/default/en.json [hunk 3]
```diff
@@ -207,6 +207,10 @@
     "defaultMessage": "{timeSince, plural, =1 {1 minute} other {# minutes}} ago",
     "description": "Text for time in minutes since given date for MLflow views"
   },
+  "7kUi8J": {
+    "defaultMessage": "Duration:",
+    "description": "Row title for the duration of runs on the experiment compare runs page"
+  },
```
Comment (sunishsheth2009) [inline mlflow/server/js/src/i18n/default/en.json:213]: Might have to run the i18n extract script again to add `unknown` to the default strings here

**Pull request merged**